src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,50 @@
 
 class Animal {
   // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
+  static alive = [];
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  // constructor(name) {
+  //   super(name);
+  // }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
```

Comment:
The `bite` method checks if the target is an instance of `Herbivore` and not hidden. It would be safer to also check if the target is an instance of `Animal` to ensure the target has a `health` property.